pkg/k8splan/watcher.go

```diff
@@ -221,6 +307,27 @@ func (w *watcher) start(ctx context.Context) {
 	}
 }
 
+func (w *watcher) updateSecret(core corecontrollers.Interface, secret *v1.Secret) (*v1.Secret, error) {
+	var resultingSecret *v1.Secret
+	err := retry.OnError(retry.DefaultBackoff,
+		func(err error) bool {
+			if apierrors.IsConflict(err) {
```

Comment:
To be clear, we're not retrying if it's a conflict, but we will for other errors?

Reply:
Correct. This is a request that is originally from Darren, and will handle transient errors such as GOAWAY or server unavailable etc. 

The only time we will guaranteed not be able to update is if there is a conflict as the server will never take our version, so in that case we scrap what we're trying to do